game/src/info/trip.rs

```diff
@@ -386,6 +385,328 @@ pub fn cancelled(ctx: &mut EventCtx, app: &App, id: TripID) -> Widget {
+                ))
+                .bg(bg_color)
+                .render(ctx)
+                .scale(0.4)
+                .centered_on(pt),
+            );
+        }
+    }
+}
+
+/// Helper func for make_bar()
+/// Builds a default text overlay widget
+fn build_text(msgs: &Vec<String>, distance: &Distance) -> Text {
+    let mut display_txt = Text::new();
+    for msg in msgs {
+        display_txt.add(Line(msg));
+    }
+    display_txt.add(Line(format!(
+        "  Distance covered: {}",
+        distance.describe_rounded()
+    )));
+    display_txt
+}
+
+/// Makes the timeline bar in trip info panel
+fn make_bar(
+    ctx: &mut EventCtx,
+    app: &App,
+    trip_id: TripID,
+    phases: &Vec<TripPhase>,
+    progress_along_path: Option<f64>,
+) -> Widget {
+    let map = &app.primary.map;
+    let blank_lane_speeds_bt = BTreeMap::new();
+    let blank_intersection_delays_bt = BTreeMap::new();
+    let intersection_delays = &app
+        .primary
+        .sim
+        .get_analytics()
+        .trip_intersection_delays
+        .get(&trip_id)
+        .unwrap_or(&blank_intersection_delays_bt);
+    let lane_speeds = &app
+        .primary
+        .sim
+        .get_analytics()
+        .lane_speed_percentage
+        .get(&trip_id)
+        .unwrap_or(&blank_lane_speeds_bt);
+    let box_width = 0.22 * ctx.canvas.window_width;
+    let mut total_dist = Distance::meters(0.0);
+    let mut segments: Vec<(Distance, Color, Text)> = Vec::new();
+    let trip = app.primary.sim.trip_info(trip_id);
+    let end_time = phases.last().as_ref().and_then(|p| p.end_time);
+    let total_duration_so_far = end_time.unwrap_or_else(|| app.primary.sim.time()) - trip.departure;
+
+    let mut icons_geom = Vec::new();
+    let mut icon_pos = Vec::new();
+    for (idx, p) in phases.into_iter().enumerate() {
+        let mut msgs = Vec::new();
+        let norm_color = color_for_trip_phase(app, p.phase_type).alpha(0.7);
+        msgs.push(p.phase_type.describe(map));
+        msgs.push(format!("  Started at {}", p.start_time.ampm_tostring()));
+        let phase_duration = if let Some(t2) = p.end_time {
+            let d = t2 - p.start_time;
+            msgs.push(format!(
+                "  Ended at {} (duration: {})",
+                t2.ampm_tostring(),
+                d
+            ));
+            d
+        } else {
+            let d = app.primary.sim.time() - p.start_time;
+            msgs.push(format!("  Ongoing (duration so far: {})", d));
+            d
+        };
+        // TODO Problems when this is really low?
+        let percent_duration = if total_duration_so_far == Duration::ZERO {
+            0.0
+        } else {
+            phase_duration / total_duration_so_far
+        };
+
+        let phase_width = box_width * percent_duration;
+        if idx == phases.len() - 1 {
+            if let Some(p) = progress_along_path {
+                icons.push(Widget::draw_batch(
```

Comment:
Build break, I think this has to be `icons_geom` now?

Reply:
Actually, I think just declaring `let mut icons = Vec::new()` up by the others should fix it. `icons_geom` is for the tooltips.